docs/proposals/20210614-podunavailablebudget.md

```diff
@@ -0,0 +1,193 @@
+3. The application owner and middleware team release the same Pods at the same time based on OpenKruise cloneSet, sidecarSet in-place upgrades
+
+## Proposal
+PodUnavailableBudget is an attempt for voluntary disruption, which is not only compatible with Kubernetes PDB protection for Eviction API, but also able to support the protection ability of above scenarios.
+### API Definition
+```
+// PodUnavailableBudgetSpec defines the desired state of PodUnavailableBudget
+type PodUnavailableBudgetSpec struct {
+	// Selector label query over pods managed by the budget
+	Selector *metav1.LabelSelector `json:"selector,omitempty"`
+
+	// TargetReference contains enough information to let you identify an workload for PodUnavailableBudget
+	// Selector and TargetReference are mutually exclusive, TargetReference is priority to take effect.
+    // Selector is commonly used in scenarios where applications are deployed using multiple workloads,
+    // and TargetReference is used for protection against a single workload.
+	TargetReference *TargetReference `json:"targetRef,omitempty"`
+
+	// Delete pod, evict pod or update pod specification is allowed if at most "maxUnavailable" pods selected by
+	// "selector" or "targetRef"  are unavailable after the above operation for pod.
+	// MaxUnavailable and MinAvailable are mutually exclusive, MaxUnavailable is priority to take effect
+	MaxUnavailable *intstr.IntOrString `json:"maxUnavailable,omitempty"`
+
+	// Delete pod, evict pod or update pod specification is allowed if at least "minAvailable" pods selected by
+	// "selector" or "targetRef" will still be available after the above operation for pod.
+	MinAvailable *intstr.IntOrString `json:"minAvailable,omitempty"`
+}
+
+// TargetReference contains enough information to let you identify an workload for PodUnavailableBudget
+type TargetReference struct {
+	// API version of the referent.
+	APIVersion string `json:"apiVersion,omitempty"`
+	// Kind of the referent.
+	Kind string `json:"kind,omitempty"`
+	// Name of the referent.
+	Name string `json:"name,omitempty"`
+}
+
+// PodUnavailableBudgetStatus defines the observed state of PodUnavailableBudget
+type PodUnavailableBudgetStatus struct {
+	// Most recent generation observed when updating this PUB status. UnavailableAllowed and other
+	// status information is valid only if observedGeneration equals to PUB's object generation.
+	// +optional
+	ObservedGeneration int64 `json:"observedGeneration"`
+
+	// DisruptedPods contains information about pods whose eviction or deletion was
+	// processed by the API handler but has not yet been observed by the PodUnavailableBudget.
+	// +optional
+	DisruptedPods map[string]metav1.Time `json:"disruptedPods,omitempty"`
+
+	// UnavailablePods contains information about pods whose specification changed(inplace-update pod),
```

Comment:
specification changes does not necessarily means inplace-update, pod container restart is also possible. plz changes to 
(e.g. inplace-update)